torch/fx/experimental/graph_gradual_typechecker.py

```diff
@@ -4,10 +4,30 @@
 from torch.fx.tensor_type import Dyn, is_consistent, TensorType, is_more_precise
 from typing import Callable, Dict
 from torch.fx.node import Target, Node
+from torch.nn.modules.batchnorm import BatchNorm2d
+
 
 _INFERENCE_RULES: Dict[Target, Callable] = {}
 
 
+def apply_matching(t, n):
```

Comment:
Maybe name this something like `expand_to_tensor_dim`? `apply_matching` isn't very descriptive IMO

Reply:
matching is the name for this operation in gradual types but I can rename it!